app/assets/javascripts/admin/addon/components/site-settings/uploaded-image-list.js

```diff
@@ -1,16 +1,24 @@
-import { action } from "@ember/object";
 import Component from "@ember/component";
-import showModal from "discourse/lib/show-modal";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import UploadedImageListModal from "admin/components/modal/uploaded-image-list";
 
 export default class UploadedImageList extends Component {
+  @service modal;
+
   @action
   showUploadModal({ value, setting }) {
-    showModal("admin-uploaded-image-list", {
-      admin: true,
-      title: `admin.site_settings.${setting.setting}.title`,
-      model: { value, setting },
-    }).setProperties({
-      save: (v) => this.set("value", v),
+    this.modal.show(UploadedImageListModal, {
+      model: {
+        title: `admin.site_settings.${setting.setting}.title`,
+        save: this.valueChanged,
```

Comment:
minor, but I'd name it `changeValue`/`setValue`

(`valueChanged` could be a function that takes a callback which is called whenever value is changed 😛)

…or even better:

```suggestion
        save: (value) => this.set("value", value),
```

No name at all! 😄